### DISCUSSION TITLE
Why does the response time take longer during edge runtime cold boot?

### DISCUSSION THREAD

**Discussion by Eunkyung-Son** [vercel/next.js#62429] [Help]
### Summary

In the document [here](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes#runtime-differences), It's stated that the edge runtime has a lower delay compared to the serverless runtime during cold starts. However, when tested directly, the serverless runtime is approximately three times faster.

Below, I duplicated two pages using the same code and configured the options differently.
<img width="1086" alt="스크린샷 2024-02-23 오전 11 07 23" src="https://github.com/vercel/next.js/assets/68625841/dc887083-8c96-49be-bc73-bcf0df2c5080">

<img width="392" alt="스크린샷 2024-02-23 오후 4 28 21" src="https://github.com/vercel/next.js/assets/68625841/b79a1527-6049-4062-8628-e9e4aff79e50">

I also set the serverless function region to icn1(South Korea) in the Vercel settings.

Below is a comparison of response time difference between edge and serverless configurations during cold boot.
During a 'cold start', it can be observed that the response time in the edge environment is longer compared to the serverless environment.
<img width="1706" alt="스크린샷 2024-02-22 오후 12 30 45" src="https://github.com/vercel/next.js/assets/68625841/21236e1f-ebea-40ee-89bf-032d60b3616f">

Below illustrates the difference in response time between pages configured with edge and serverless setups when in a warm state.
In the test results, it was evident that, on average, the edge setup exhibited faster response times.
<img width="856" alt="스크린샷 2024-02-22 오후 12 31 02" src="https://github.com/vercel/next.js/assets/68625841/1cecc15a-5d38-4b18-9667-7e7c7fbe9d18">

There seems to be no issue here. However, I'm unsure why the response time is slower in the edge setup during cold boot. Is there a specific reason for this? For your information, the backend server region is South Korea, Seoul.

**Comment by programmersEmperor** [comment]
The same thing happened to me, here I have 2 functions:
```
// Serverless Function
import { NextResponse } from "next/server";

export async function GET(){
    return NextResponse.json({'message': 'serverless finished'}); 
} 
```

```
// Edge Function
import { NextResponse } from "next/server";
export const runtime = 'edge';

export async function GET(){
    return NextResponse.json({'message': 'edge finished'}); 
}
```

When I tried measuring the response time of both of them using Postman, I found that Serverless Function Responded  faster than the Edge Function. That is weird!!!
